Sources/RoutingKit/PrefixGroupable.swift

```diff
@@ -0,0 +1,36 @@
+public protocol PrefixGroupable: Router {
+    associatedtype RouterType: PrefixGroupable
+    func grouped(_ prefix: [PathComponent]) -> RouterType
+}
+
+
+public class PrefixGroupedRouter<Output, RouterType: Router> where Output == RouterType.Output {
```

Comment:
I don't know if this has been discussed before, but is there a preferred way to conform generic types in the `vapor` org? Because if I wrote this, the `RouterType` type would be conformed in the `where` clause:

```swift
public class PrefixGroupedRouter<Output, RouterType> where RouterType: Router, Output == RouterType.Output {}
```

Reply:
This is no longer relevant to this PR, but if you'd like to add something to the style guide, leave a comment https://github.com/vapor/vapor/pull/1876. Trying to get that merged too.